src/api/entities/CorporateBallot/index.ts

```diff
@@ -116,9 +116,13 @@ export class CorporateBallot extends Entity<UniqueIdentifiers, HumanReadable> {
   public async exists(): Promise<boolean> {
     const { id, asset, context } = this;
 
-    const exists = await getCorporateBallotDetails(asset, id, context);
+    try {
+      await getCorporateBallotDetailsOrThrow(asset, id, context);
+    } catch (error) {
+      return false;
+    }
```

Comment:
Its better to have two separate util methods one returning details and other throwing error, instead of adding try catch here